include/inexor/vulkan-renderer/camera.hpp

```diff
@@ -40,6 +39,28 @@ class Camera {
         bool down{false};
     } m_keys;
 
+public:
+    Camera() = default;
+    /// @brief Constructs a camera based on camera properties.
+    /// @param type The type of the camera.
+    /// @param position The camera's position.
+    /// @param rotation The camera's rotation.
+    /// @param movement_speed The movement speed of the camera.
+    /// @param rotation_speed The rotation speed of the camera.
+    /// @param fov The camera's field of view.
+    /// @param z_near The near clipping value.
+    /// @param z_far The far clipping value.
+    /// @param window_width The width of the window which is used to calculate aspect ratio.
+    /// @param window_height The height of the window which is used to calculate aspect ratio.
+    Camera(CameraType type, glm::vec3 position, glm::vec3 rotation, float movement_speed, float rotation_speed,
```

Comment:
To resolve `[misc-non-private-member-variables-in-classes]`, I wrapped everything into a nice constructor. The code before that was mimicing the behaviour of a constructor in the first place:

```cpp
// All this code war wrapped into a constructor. Look how this setup code mimics the constructor!
m_game_camera.m_type = Camera::CameraType::LOOKAT;
Camera new_camera(CameraType::LOOKAT, {0.0f, 0.0f, 5.0f}, {0.0f, 0.0f, 0.0f}, 0.1f, 0.25f, 45.0f, 0.1f, 256.0f,
m_game_camera.m_rotation_speed = 0.25f;
m_window->width(), m_window->height());
m_game_camera.m_movement_speed = 0.1f;
m_game_camera.set_position({0.0f, 0.0f, 5.0f});
m_game_camera = new_camera;
m_game_camera.set_rotation({0.0f, 0.0f, 0.0f});
m_game_camera.set_perspective(45.0f, static_cast<float>(m_window->width()) / static_cast<float>(m_window->height()), 0.1f, 256.0f);
```